openedx-unsupported/configuration
# [SE-4612] Update elasticsearch configurations

## Description

Due to the high load on the server when provisioning new instances, elasticsearch often takes longer to startup than the default timeout defined in it's systemd service file. This leaves it in a `failed` state and requires a manual `systemctl restart` to get it running.

Also, the elasticsearch configurations applied in the ansible role are being ignored because of a change in env vars (`ES_CONF` -> `ES_PATH_CONF`), and also wouldn't work with recent ES versions even if they were applied.

In the first commit I replaced the ES configuration files in the ansilbe role with ones that would work with more recent version of ES. On first glance there may seem to be a lot of changes in the configurations but it would probably help to know where I got these configurations from:
- I installed elasticsearch v7.10 on a fresh ubuntu server
- I took the configuration files that came with that installation (`/etc/elasticsearch/...`) and placed them instead of the ones we had in the playbook templates
- I changed the templates to use the variables defined in `playbooks/roles/elasticsearch/defaults/main.yml` in the relevant places of the new configuration files

In the second commit I increased the timeout for starting the ES systemd service.

## Supporting information

**Sandbox Link**: [SE-4612: Discussion Forum / ES Configs](https://manage.opencraft.com/instance/28910/edx-appserver/19959/)
**Sandbox Server IP**: `213.32.74.63`

## Testing instructions

Since this was an issue that occurred when configuring new servers, you could test these on a new server to make sure everything works fine. However the sandbox I've provided is a freshly configured one without any manual intervention so it should suffice to check that.

#### Testing by using the forums
We can test that elasticsearch is running be testing the discussion forums, since this forums won't work if ES is not running:
- Go to the [course](https://se-4612.opencraft.hosting/courses/course-v1:edX+DemoX+Demo_Course/course/) page on the sandbox instance
- Click on the 'Discussions' link
- The page should open without errors, creating and searching for posts should work

#### Testing on the server
If you have SSH access to the server (I can give you access if you need it):

- Check that the elasticsearch service is running (`systemctl status elasticsearch`)
- Check that the elasticsearch process is running with the right configurations. To do this check it's enough to check elasticsearch is running with a `512m` heapsize rather than the default `1g`.  You can check this by running the command below:
    ```
    ps aux | grep elasticsearch | awk 'match($0,/-Xmx[0-9]+m/) {print substr($0,RSTART,RLENGTH)}'
    ```
    If it outputs `-Xmx512m` then we're good, if it outputs `-Xmx1g` then it's using the wrong configurations.
